### PR TITLE
Port build to SBT 1.x

### PR DESCRIPTION
**Pull request by 2m**
Mostly plugin updates.

Sbt 1.x prints out a log message for every build.sbt file it loads. So there was a lot of log output when Akka build was loading. Therefore I have also merged everything to one build file (like it is done in Alpakka, which is more "modern" build :). This has also uncovered quite a bit of settings duplication which was removed.

Refs #23093

### CODE REVIEW

**Comment by ktoso** [comment]
> Sbt 1.x prints out a log message for every build.sbt file it loads. So there was a lot of log output when Akka build was loading. Therefore I have also merged everything to one build file (like it is done in Alpakka, which is more "modern" build :). This has also uncovered quite a bit of settings duplication which was removed.

That's very good, I like to have it in 1 file.

File: project/MultiNode.scala [hunk 1]
```diff
@@ -69,11 +69,14 @@ object MultiNode extends AutoPlugin {
...
-                multiNodeResults.overall
-              else
-                testResults.overall
+            //        // FIXME no idea what this was doing
+            //        val overall =
+            //          if (testResults.overall.id < multiNodeResults.overall.id)
+            //            multiNodeResults.overall
+            //          else
+            //            testResults.overall
+            val overall = testResults.overall
```
Comment (ktoso) [inline project/MultiNode.scala]: did it break if we wanted to keep it? I believe this makes the tests fail when multinode tests fail so we may need to keep it.

File: project/ParadoxBrowse.scala [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+  paradoxBrowse := {
+    import java.awt.Desktop
+    val rootDocFile = (target in (Compile, paradox)).value / "index.html"
+    val log = streams.value.log
+    if (!rootDocFile.exists()) log.info("No generated docs found, generate with the 'paradox' task")
+    else if (Desktop.isDesktopSupported) Desktop.getDesktop.open(rootDocFile)
+    else log.info(s"Couldn't open default browser, but docs are at $rootDocFile")
+  }
+}
```
Comment (ktoso) [inline project/ParadoxBrowse.scala]: yay

Comment (ktoso) [reply project/ParadoxBrowse.scala]: Mention it in CONTRIBUTING?

File: project/Publish.scala [hunk 3]
```diff
@@ -25,12 +25,6 @@ object Publish extends AutoPlugin {
     defaultPublishTo := crossTarget.value / "repository")
 
   def akkaPomExtra = {
-    /* The scm info is automatic from the sbt-git plugin
-    <scm>
-      <url>git://github.com/akka/akka.git</url>
-      <connection>scm:git:git@github.com:akka/akka.git</connection>
-    </scm>
-    */
```
Comment (ktoso) [inline project/Publish.scala]: good

File: project/VersionGenerator.scala [hunk 4]
```diff
@@ -9,9 +9,9 @@ import sbt.Keys._
 /**
  * Generate version.conf and akka/Version.scala files based on the version setting.
  */
-object Version {
+object VersionGenerator {
```
Comment (ktoso) [inline project/VersionGenerator.scala]: good, just Version would conflict with some things included from plugins if i remember correctly

File: project/build.properties [hunk 5]
```diff
@@ -1 +1 @@
-sbt.version=0.13.16
+sbt.version=1.0.2
```
Comment (ktoso) [inline project/build.properties]: 🎆 🎆 🎆 🎆 🎆

File: project/plugins.sbt [hunk 6]
```diff
@@ -1,45 +1,24 @@
...
-addSbtPlugin("com.lightbend.akka" % "sbt-paradox-akka" % "0.3")
-
+addSbtPlugin("com.typesafe.sbt" % "sbt-native-packager" % "1.2.2-RC2")
+addSbtPlugin("io.spray" % "sbt-boilerplate" % "0.6.1")
+addSbtPlugin("com.timushev.sbt" % "sbt-updates" % "0.3.1")
+addSbtPlugin("com.lightbend.akka" % "sbt-paradox-akka" % "0.4")
 addSbtPlugin("com.lightbend" % "sbt-whitesource" % "0.1.6")
-
 addSbtPlugin("com.typesafe.sbt" % "sbt-git" % "0.9.3")
+addSbtPlugin("net.virtual-void" % "sbt-dependency-graph" % "0.9.0") // for advanced PR validation features
```
Comment (ktoso) [inline project/plugins.sbt]: :-)

**Review by ktoso — APPROVED** [review]
LGTM, though watch out about the change in multi-jvm – i think that one has to be changed back

**Comment by ktoso** [comment]
This one likely has to be kept/reverted: https://github.com/akka/akka/pull/23850#discussion_r147020087

File: project/MultiNode.scala (hunk 1 above)
Comment (2m) [reply project/MultiNode.scala]: Will use this to make it work: https://github.com/akka/akka-http/pull/1435/files#r147072897

**Comment by ktoso** [comment]
LGTM, ship it! :)

**Review by johanandren — APPROVED** [review]
LGTM! Tried it out locally, ran some tests both multi-jvm and regular ones, generated docs didn't notice any hiccups.

**Review by ktoso — APPROVED** [review]

**Comment by ktoso** [comment]
Wohoo!

**Pull request merged**